akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/BehaviorTestKitSpec.scala

```diff
@@ -395,6 +396,20 @@ class BehaviorTestKitSpec extends AnyWordSpec with Matchers with LogCapturing {
       val newChild = testkit.expectEffectType[Spawned[_]]
       child.childName shouldBe newChild.childName
     }
+
+    "start and stop child and then stop testkit itself" in {
+      val testkit = BehaviorTestKit(Parent.init)
+      testkit.run(SpawnChild)
+      val child = testkit.expectEffectType[Spawned[String]]
+      testkit.run(StopChild(child.ref))
+      testkit.expectEffect(Stopped(child.childName))
```

Comment:
This part of the test is exactly the same as in "stop and restart a named child". Do we have to repeat that test here? How is that related to the `stopSelf`?

Reply:
The difference is the hasEffects, the new pr has added the stop effect of the current actor in the stopSelf method.